src/main/java/org/checkerframework/checker/dependencyinjection/DependencyInjectionAnnotatedTypeFactory.java

```diff
@@ -1,18 +1,164 @@
+            "to",
+            processingEnv,
+            "com.google.inject.TypeLiteral<? extends T>"));
+    this.toMethods.add(
+        TreeUtils.getMethod(
+            "com.google.inject.binder.LinkedBindingBuilder",
+            "to",
+            processingEnv,
+            "com.google.inject.Key<? extends T>"));
     super.postInit();
   }
 
+  /* Returns true iff the argument is an invocation of AbstractModule.bind.
+   *
+   * @param methodTree the method invocation tree
+   * @return true iff the argument is an invocation of AbstractModule.bind()
+   */
+  private boolean isBindMethod(Tree methodTree) {
+    return TreeUtils.isMethodInvocation(methodTree, this.bindMethods, this.getProcessingEnv());
+  }
+
+  /* Returns true iff the argument is an invocation of AbstractModule.to() */
+  private boolean isToMethod(Tree methodTree) {
+    return TreeUtils.isMethodInvocation(methodTree, this.toMethods, this.getProcessingEnv());
+  }
+
   @Override
   protected void postAnalyze(ControlFlowGraph cfg) {
+
+    System.out.println("--------------------");
+
+    Set<Block> visited = new HashSet<>();
+    Deque<Block> worklist = new ArrayDeque<>();
+    HashMap<Integer, Integer> knownBindings = new HashMap<>();
+
+    Block entry = cfg.getEntryBlock();
+    worklist.add(entry);
+    visited.add(entry);
+
+    while (!worklist.isEmpty()) {
+      Block current = worklist.remove();
+
+      current
+          .getNodes()
+          .forEach(
+              node -> {
+                if (node instanceof MethodInvocationNode) {
+                  MethodInvocationNode methodInvocationNode = (MethodInvocationNode) node;
+
+                  System.out.printf("Method Invocation: %s\n", methodInvocationNode);
+                  System.out.printf("Operands: %s\n\n", methodInvocationNode.getOperands());
+
+                  if (methodInvocationNode.getOperands().size() >= 2) {
+                    MethodAccessNode methodAccessNode = methodInvocationNode.getTarget();
+                    Node fieldAccessNode = methodInvocationNode.getArgument(0);
+
+                    if (isBindMethod(methodInvocationNode.getTree())) {
+                      // Class that is being bound - put in knownBindings
+                      AnnotatedTypeMirror boundClassTypeMirror =
+                          this.getAnnotatedType(fieldAccessNode.getTree());
+
+                      knownBindings.put(boundClassTypeMirror.getUnderlyingTypeHashCode(), null);
+                    } else if (isToMethod(methodInvocationNode.getTree())) {
+                      Node receiver = methodAccessNode.getReceiver();
+
+                      // TODO: This feels a little hacky to get the bound class type mirror
+                      // Class that is being bound - should be in knownBindings
+                      AnnotatedTypeMirror boundClassTypeMirror =
+                          this.getAnnotatedType(
+                              receiver.getOperands().toArray(new Node[2])[1].getTree());
```

Comment:
This is hacky - you're making some very strong assumptions about the underlying program structure that are unlikely to be true in general. I suspect that we need to come up with some kind of multiple stage analysis here: for example, the call to `bind()` puts us into some intermediate "binding in progress" state, which is eventually resolved by the call to `to()`.